vme/zone/spells.zon

```diff
@@ -14880,6 +14891,29 @@ code
 }
 dilend
 
+/*
+New function for handling bonus calculations.
+Uses the old code as a starting point for bonuses of up to 10 points but removes the cap,
+allowing bonuses over 10 if the roll is high enough.
+Remember that rolls for attribute boosts are generally the sum of the characters skill and
+relevant attribute + an open100 roll -50.
+Example: Character with 50 magic and 50 skill and rolls a 50 would have a total of 100.
+1-10 are roughly 1 point per 10 points on your roll
+In this case, this function would return 10, the old standard roll.
+However, if the same character were to roll a 160 (due to how open roll works. If you roll a 100 it rolls again and adds the result so here I'm assuming they rolled a 100, then a 10) then the after the first 10 points, the second part of the calculation would kick in and the result would be an 12.
+*/
+dilbegin integer bonusCalc(hm : integer);
+
+code
+{
+      hm := hm/10+1;  // old starting calculation
+      if (hm>10)
+	hm := 10+((hm-10)/3); // allows for numbers greater than 10 at intervals of 30 per increase
```

Comment:
Isn't this supposed to be / 30 ? (or /20)

Reply:
hm is first divided by 10 (and then 1 is added to be consistent with old behavior).
then, if that is greater than 10, we further divide it by 3.

eg:
      16 = 150/10+1;  // old starting calculation
      if (16>10)
	12 := 10+((16-10)/3); // allows for numbers greater than 10 at intervals of 30 per increase

so this means that a roll of 150 gives you a +12 (based on our previous discussions. Was originally /5 which would mean it was a +11 instead)